eZ/Publish/Core/Persistence/Legacy/Content/UrlAlias/Gateway.php

```diff
@@ -227,4 +227,12 @@
      * @return int
      */
     abstract public function getLocationContentMainLanguageId($locationId);
+
+    /**
+     * Removes languageId of removed translation from lang_mask and deletes single language rows for multiple Locations.
+     *
+     * @param int $languageId Language Id to be removed
+     * @param string[] $actions actions for which to perform the update
+     */
+    abstract public function bulkRemoveTranslation($languageId, $actions);
```

Comment:
There is already [protected method named `removeTranslation`](https://github.com/ezsystems/ezpublish-kernel/blob/8c961a2976e34e768aeb99f619c9043020908c01/eZ/Publish/Core/Persistence/Legacy/Content/UrlAlias/Gateway/DoctrineDatabase.php#L430) which relies on a different set of parameters. Reusing that would require re-fetching data and wouldn't cover all the described cases, so I've decided to create my own method.

`bulk` refers to multiple actions which internally refer to multiple Locations of a Content Object.